src/Microsoft.ML.FastTree/GamRegression.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.CommandLine;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.FastTree;
+using Microsoft.ML.Runtime.FastTree.Internal;
+using Microsoft.ML.Runtime.Internal.Internallearn;
+using Microsoft.ML.Runtime.Model;
+using Microsoft.ML.Runtime.Training;
+
+[assembly: LoadableClass(RegressionGamTrainer.Summary,
+    typeof(RegressionGamTrainer), typeof(RegressionGamTrainer.Arguments),
+    new[] { typeof(SignatureRegressorTrainer), typeof(SignatureTrainer), typeof(SignatureFeatureScorerTrainer) },
+    RegressionGamTrainer.UserNameValue,
+    RegressionGamTrainer.LoadNameValue,
+    RegressionGamTrainer.ShortName, DocName = "trainer/GAM.md")]
+
+[assembly: LoadableClass(typeof(RegressionGamPredictor), null, typeof(SignatureLoadModel),
+    "GAM Regression Predictor",
+    RegressionGamPredictor.LoaderSignature)]
+
+namespace Microsoft.ML.Runtime.FastTree
+{
+    public sealed class RegressionGamTrainer :
+    GamTrainerBase<RegressionGamTrainer.Arguments, RegressionGamPredictor>
+    {
+        public partial class Arguments : ArgumentsBase
+        {
+            [Argument(ArgumentType.AtMostOnce, HelpText = "Metric for pruning. (For regression, 1: L1, 2:L2; default L2)", ShortName = "pmetric")]
+            [TGUI(Description = "Metric for pruning. (For regression, 1: L1, 2:L2; default L2")]
+            public int PruningMetrics = 2;
```

Comment:
>int PruningMetrics = 2 [](start = 19, length = 22)

wow, really? Can we maybe have enum here?

Reply:
This is a holdover from `FastTree` -- this is copied over from `FastTreeRegression`.  It's a parameter to the [`RegressionTest` class](https://github.com/dotnet/machinelearning/blob/a8cd341f8c03985aaea216e8c911c466ad020b1e/src/Microsoft.ML.FastTree/Training/Test.cs#L516).

If possible, I'd like to file this as a separate issue and fix as a separate PR.